ccdaservice/serveccda.js

```diff
@@ -26,6 +26,39 @@ var webRoot = "";
 var authorDateTime = '';
 var documentLocation = '';
 
+class DataStack {
+    constructor(delimiter) {
+        this.delimiter = delimiter;
+        this.buffer = "";
+    }
+
+    endOfCcda() {
+        return this.buffer.length === 0 || this.buffer.indexOf(this.delimiter) === -1;
+    }
+
+    pushToStack(data) {
+        this.buffer += data;
+    }
+
+    fetchBuffer() {
+        const delimiterIndex = this.buffer.indexOf(this.delimiter);
+        if (delimiterIndex !== -1) {
+            const bufferMsg = this.buffer.slice(0, delimiterIndex);
+            this.buffer = this.buffer.replace(bufferMsg + this.delimiter, "");
```

Comment:
couldn't you get the same effect by just doing 
```this.buffer = "";```
Not sure why you need the this.buffer.replace here.

Reply:
In current use case this may be true.  But if I wanted to read the buffer before delimit, I don't want to destroy stack.
This made sense to me. Do you see a potential issue?

Reply:
So you've already grabbed the bufferMsg in line 46.  Line 47, if I'm reading it correctly just sets this.buffer to be an empty string.  I didn't see why you need the replace operation when you can just assign this.buffer = ""; on line 47.

Reply:
Just because I'm not currently validating data doesn't mean I won't in future. The intent is to be able to read the current stack at anytime but not destroy stack until the end of file is received.
Being lazy, I probably should have added a new function to scan buffer to avoid confusion.
I could add if you feel strongly

Reply:
Say I want to fetch by different delimiter like new line and process received data line by line. I could just return out of receive to a function that pulls from the stack line by line by resetting delimiter from eof to new line i.e FIFO.
Even though this is not the way it is used currently doesn't mean it won't be in future either here or a different workflow using service.